core/database/foxx/api/user_router.js

```diff
@@ -516,9 +519,47 @@ router.get('/token/set', function(req, res) {
                         refresh: req.queryParams.refresh,
                         expiration: Math.floor(Date.now() / 1000) + req.queryParams.expires_in
                     };
-                    g_db._update(user_id, obj, {
-                        keepNull: false
-                    });
+                    if (token_type && other_token_data) {
+
+                        // NOTE: the edge is only defined for the `u` collection and the `globus_coll` collection.
```

Comment:
Upon another look, it seems like `getUserFromClientID` will check against edge(s) to ensure there is a link between either a `uuid` or a domain account identity to a user account `u`, so this should either be replaced with simply using `user_id` above, or because we need to always have access to the key, `const user_doc = g_db.u.document(user_id);`